src/com.spoiledcat.git.ui/Editor/UI/HierarchyWindowInterface.cs

```diff
@@ -0,0 +1,64 @@
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Unity.VersionControl.Git
+{
+    [InitializeOnLoad]
+    public class HierarchyWindowInterface
+    {
+        static HierarchyWindowInterface()
+        {
+            EditorApplication.hierarchyWindowItemOnGUI += OnHierarchyItemTryToDrawStatusIcon;
+        }
+
+        private static void OnHierarchyItemTryToDrawStatusIcon(int instanceID, Rect selectionRect)
+        {
+            string guid;
+            GameObject hierarchyGO = EditorUtility.InstanceIDToObject(instanceID) as GameObject;
+            if (!hierarchyGO)
+            {
+                // if no Object has been returned by the InstanceIDToObject() method, then it is possible, that it is a scene
+                string scenePath = "";
+                for (int i = 0; i < SceneManager.sceneCount; i++)
+                {
+                    Scene scene = SceneManager.GetSceneAt(i);
+                    if (scene.GetHashCode() == instanceID)
+                    {
+                        scenePath = scene.path;
+                        break;
+                    }
+                }
+
+                if (string.IsNullOrEmpty(scenePath))
+                    return;
+
+                guid = AssetDatabase.AssetPathToGUID(scenePath);
+            }
+            else
+            {
+                if (hierarchyGO != PrefabUtility.GetNearestPrefabInstanceRoot(hierarchyGO))
+                    return;
+
+                GameObject prefab = PrefabUtility.GetCorrespondingObjectFromSource(hierarchyGO);
+                if (!prefab)
+                    return;
+
+                if (!AssetDatabase.TryGetGUIDAndLocalFileIdentifier(prefab, out guid, out long localId))
+                    return;
+            }
+
+
+            Texture2D texture = ProjectWindowInterface.GetStatusIconForAssetGUID(guid);
+            if (texture == null)
+                return;
+
+            // place the icon to the right of the list:
+            Rect r = new Rect(selectionRect);
+            r.x = r.width + 10;
```

Comment:
The icons seem to be aligned with an inverse indentation to the list item indentation, instead of being all aligned to the right of the window regardless of the indentation of the list entry. That looks odd? This probably needs to be `r.x = r.xMax - 10`, so it's aligned to the right no matter how much indentation the entry has.

Reply:
I've been using this tool for a while now in my own project and I kinda got used to that incidental indentation - it's easier to see which Hierarchy Icon belongs to which prefab object, when those Hierarchy Icons mimick the indentation of the prefab objects. But, I understand that it might not be for everyone, so I added an option to the `SettingsView` to turn on/off that indentation, I hope it is okay :)